### DISCUSSION TITLE
UAV Modelling - diffraction

### DISCUSSION THREAD

**Discussion by Advaith2212** [NVlabs/sionna#697] [General]
We are modelling considering an empty scene with TX, RX and a UAV (`itu_metal`). 

We want to check if UAV diffracts and generates a path between TX and RX (`iso` in both case). 

In most locations of UAV, there seems to be no paths. Only for certain positions of the UAV, there seems to be a path contributed by the UAV.
 
We have also increased the `num_samples` parameter to `1e8` range. 

We would like to know if there is any criteria for the object in a scene to diffract.

Thanks and regards

![Screenshot 2024-12-11 112143](https://github.com/user-attachments/assets/3368e75e-cbf5-4195-84b0-26f62406a9d8)

![image](https://github.com/user-attachments/assets/709253b4-906b-4e56-a80f-5a408ff0b8d4)

**Comment by jhoydis** [comment]
Hi @Advaith2212

I assume that you are not explicitly asking for diffracted paths, but rather why there are no paths at all.

1) Your UAV models looks very complex and is probably composed of many very small triangles. It would probably make sense to simplify the UAV geometry to a box or something like this.

2) If the distance between the TX and UAV is large, you need to shoot many rays to actually hit the UAV. If the number of rays is too small,  the UAV will not be "seen" at all by the ray tracer.

3) Have you tried making the UAV a diffuse reflector (i.e., enable scattering)?

**Comment by Advaith2212** [comment]
Greetings,

I simplified the model assumed for `UAV` and tried the same. It seems to be pretty random most of the times. 

Our scenario was as follows : 

1. Empty Scene with `TX-RX` distance : `40m` and are assumed to be at same height
2. UAV is located near the RX 
3. UAV size is (`1.6m x 1.5m x 0.7m`) 
4. Diffraction and reflection is turned `on` 

 `Case-1) UAV is moved along XYZ co-ordinates wrt the RX`
![Screenshot from 2025-01-08 11-01-00](https://github.com/user-attachments/assets/cd979b79-eaef-45d4-86d6-281bc4b65095) 
`Case-2) UAV is moved along Y axis wrt RX`
![Screenshot from 2025-01-08 10-56-40](https://github.com/user-attachments/assets/1af3522d-661c-4fa9-8e09-3bf37b6767cc)
`Case-3) UAV is moved along XY axis wrt RX`
![Screenshot from 2025-01-08 11-05-46](https://github.com/user-attachments/assets/3e62b65e-f05a-4a62-b696-9b5b3f38bc3d)

We are confused as to why in Case-1 even though`UAV` is closer to the RX, there seems to be no paths contributed by it, whereas in Case-3, it is very far but it still does. 

I have enclosed my scene file in this discussion for reference. 
[drone_ball_1.6_1.5_0.7_new.zip](https://github.com/user-attachments/files/18342386/drone_ball_1.6_1.5_0.7_new.zip)

Can you kindly help us understand in this regard.
